napari_clusters_plotter/_kmeans_clustering.py

```diff
@@ -1,158 +1,155 @@
+        self.kmeans_settings_container_iter.layout().addWidget(QLabel("Number of Iterations"))
+        self.kmeans_nr_iterations = create_widget(widget_type="SpinBox",
+                                                  name='kmeans_nr_iter',
+                                                  value=DEFAULTS['kmeans_nr_iterations'],
+                                                  options=dict(min=1, max=10000))
+
+        self.kmeans_settings_container_iter.layout().addWidget(self.kmeans_nr_iterations.native)
+        self.kmeans_settings_container_iter.setVisible(False)
 
         # Clustering options for HDBSCAN
         # Todo
 
         # Run button
-        run_widget = QWidget()
-        run_widget.setLayout(QHBoxLayout())
-        button = QPushButton("Run")
+        run_container = QWidget()
+        run_container.setLayout(QHBoxLayout())
+        run_button = QPushButton("Run")
+        run_container.layout().addWidget(run_button)
+
+        # Update measurements button
+        update_container = QWidget()
+        update_container.setLayout(QHBoxLayout())
+        update_button = QPushButton("Update Measurements")
+        update_container.layout().addWidget(update_button)
+
+        # Defaults button
+        defaults_container = QWidget()
+        defaults_container.setLayout(QHBoxLayout())
+        defaults_button = QPushButton("Restore Defaults")
+        defaults_container.layout().addWidget(defaults_button)
+
+        # adding all widgets to the layout
+        self.layout().addWidget(title_container)
+        self.layout().addWidget(labels_layer_selection_container)
+        self.layout().addWidget(choose_properties_container)
+        self.layout().addWidget(self.clust_method_container)
+        self.layout().addWidget(self.kmeans_settings_container_nr)
+        self.layout().addWidget(self.kmeans_settings_container_iter)
+        self.layout().addWidget(run_container)
+        self.layout().addWidget(defaults_container)
+        self.layout().addWidget(update_container)
+        self.layout().setSpacing(0)
 
         def run_clicked():
 
-            if self.get_selected_label() is None:
+            if self.labels_select.value is None:
                 warnings.warn("No labels image was selected!")
                 return
 
             self.run(
-                self.get_selected_label(),
+                self.labels_select.value,
                 [i.text() for i in self.properties_list.selectedItems()],
-                self.kmeans_nr_clusters.value(),
-                self.kmeans_nr_iter.value()
+                self.kmeans_nr_clusters.value,
+                self.kmeans_nr_iterations.value
             )
 
-        button.clicked.connect(run_clicked)
-        run_widget.layout().addWidget(button)
+        run_button.clicked.connect(run_clicked)
+        update_button.clicked.connect(self.update_properties_list)
+        defaults_button.clicked.connect(partial(restore_defaults, self, DEFAULTS))
```

Comment:
I'm just thinking: If we had defaults, also for other algorithms such as GMM or HDBScan, would "DEFAULTS" still work? Or should we have "KMEANS_DEFAULTS"?

Reply:
If I read the code correctly the restoring to defaults will go through all widgets in the dictionary and change the values to the default ones. If the HDBSCAN values are also in the dictionary it should restore them simultaneously. Maybe @lazigu can confirm or deny this..

Reply:
It would! I actually tried to implement HDBScan in another branch (it does not yet function properly, but buttons are there already), so you can see that it works to restore defaults :) So my idea is that I will have for different widgets of the plugin different Defaults dictionaries, and then I am passing it to restore_defaults function in the utilities.

Reply:
That sounds great! if you need any help with HDBSCAN just let me know, maybe I have already stumbled into the same problem :)

Reply:
Ah I didn't see this message from you but I already took up on your offer :D